libcodechecker/analyze/analysis_manager.py

```diff
@@ -176,23 +198,13 @@ def signal_handler(*arg, **kwarg):
 
     signal.signal(signal.SIGINT, signal_handler)
 
-    # Remove characters which could cause directory creation problems.
-    no_spec_char_name = re.sub(r'[^\w\-_. ]', '_', args.name)
```

Comment:
This was only used to create a temporary folder, identified by the run-name, under the workspace.
Because the analysis results, from now on, are directly saved into a folder given by the user, it is no longer needed. The default value is `~/.codechecker/reports`, which is safe, and if the user specifies a custom one, it will be earlier an error, when the output folder is attempted to be created.